### PR TITLE
View All Categories Alphabetically

### LINKED ISSUE #39: View All Categories
As an admin I would like to see all the available categories so that I can choose to edit or delete one, or see that I should add a new one.

**Given** an admin is in the app  
**When** they select the `Category Management` link in the menu  
**Then** they should be directed to a page that lists all the Category names ordered alphabetically  

> **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.

### PR DESCRIPTION
**Pull request by IsaacWeiser**
#### Changes Made
1. Added file `Category.cs` to `models` directory.
2. Added file `CategoryRepository.cs` to `repositories` directory.
3. Added file `ICategoryRepository.cs` to `repositories` directory.
4. Added file `CategoryController.cs` to `controllers` directory.
5. Added file `CategoryManager.js` to `client/modules` directory.
6. Added file `Categories.js` to `client/components` directory.
7. Modified file `applicationviews` to include `category route` functionality.

#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout iw-getAllCategories
    ```
7. Open a new Terminal tab (⌘T) and navigate to the server directory.
8. Test app functionality.
    > On the nav bar select categories
    > When clicked a page of all categories ordered alphabetically should appear.
9. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

**Review by matt-rich-anderson — APPROVED** [review]
The PR appears to be working as expected.

File: Tabloid/Controllers/CategoryController.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Tabloid.Repositories;
+
+namespace Tabloid.Controllers
```
Comment (heymonicakay) [inline Tabloid/Controllers/CategoryController.cs:6]: Should this controller be auth protected? do we need any other auth checks on the route?

File: Tabloid/client/src/modules/CategoryManager.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const baseUrl = `/api/category`;
+
+export const getAllCategories = () => {
```
Comment (heymonicakay) [inline Tabloid/client/src/modules/CategoryManager.js:5]: if this route is auth protected, you'll need to get an api token to pass in to this fetch call.

File: Tabloid/client/src/components/Header.js [hunk 3]
```diff
@@ -18,36 +18,53 @@ export default function Header({ isLoggedIn }) {
   return (
     <div>
       <Navbar color="light" light expand="md">
-        <NavbarBrand tag={RRNavLink} to="/">Tabloid</NavbarBrand>
+        <NavbarBrand tag={RRNavLink} to="/">
+          Tabloid
+        </NavbarBrand>
         <NavbarToggler onClick={toggle} />
```
Comment (heymonicakay) [inline Tabloid/client/src/components/Header.js:24]: what is `toggle`?

**Comment by heymonicakay** [comment]
which issues are related to this PR?

File: Tabloid/Repositories/CategoryRepository.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.Extensions.Configuration;
+using System.Collections.Generic;
+using Tabloid.Models;
+using Tabloid.Utils;
+
+namespace Tabloid.Repositories
```
Comment (heymonicakay) [inline Tabloid/Repositories/CategoryRepository.cs:6]: Are there DbUtils methods that could be used in this repo to do some of this work? We'll want to use those utils methods consistently across the codebase.

**Review by heymonicakay — CHANGES_REQUESTED** [review]
Please see my comments

File: Tabloid/Controllers/CategoryController.cs (hunk 1 above)
Comment (IsaacWeiser) [reply Tabloid/Controllers/CategoryController.cs:6]: Taken care of @heymonicakay

File: Tabloid/client/src/modules/CategoryManager.js (hunk 2 above)
Comment (IsaacWeiser) [reply Tabloid/client/src/modules/CategoryManager.js:5]: Taken care of @heymonicakay

File: Tabloid/client/src/components/Header.js (hunk 3 above)
Comment (IsaacWeiser) [reply Tabloid/client/src/components/Header.js:24]: Toggle allows for the navbar to be open and closed when the page is shrunk. @heymonicakay

File: Tabloid/Repositories/CategoryRepository.cs (hunk 4 above)
Comment (IsaacWeiser) [reply Tabloid/Repositories/CategoryRepository.cs:6]: resolved @heymonicakay

**Review by heymonicakay — APPROVED** [review]

**Pull request merged**